SkyEditor.RomEditor.Rtdx/Domain/Common/Structures/Gyu0.cs

```diff
@@ -0,0 +1,339 @@
+﻿using SkyEditor.IO.Binary;
+using System;
+using System.IO;
+using System.Text;
+
+namespace SkyEditor.RomEditor.Domain.Common.Structures
+{
+    public static class Gyu0
+    {
+        private enum Opcode
+        {
+            Copy,
+            SplitCopy,
+            Fill,
+            Skip
+        }
+
+        public static IBinaryDataAccessor Decompress(IReadOnlyBinaryDataAccessor data)
+        {
+            var Magic = data.ReadInt32(0);
+            var Length = data.ReadInt32(4);
+
+            IBinaryDataAccessor dest = new BinaryFile(new byte[Length]);
+
+            var dataIndex = 8;
+            var destIndex = 0;
+
+            while (true)
+            {
+                byte cur = data.ReadByte(dataIndex++);
+
+                if (cur < 0x80)
+                {
+                    var next = data.ReadByte(dataIndex++);
+
+                    // EOF marker
+                    if (cur == 0x7F && next == 0xFF)
+                        return dest;
+
+                    var offset = 0x400 - ((cur & 3) * 0x100 + next);
+                    var bytes = dest.ReadSpan(destIndex - offset, (cur >> 2) + 2);
+                    dest.Write(destIndex, bytes);
+                    destIndex += bytes.Length;
+                    continue;
+                }
+
+                var op = (Opcode)((cur >> 5) - 4);
+                var arg = cur & 0x1F;
+
+                switch (op)
+                {
+                    case Opcode.Copy:
+                        {
```

Comment:
Curly braces aren't required in switch statements. C# has extra rules preventing you from shooting yourself in the foot, but switch statements are just glorified goto's.

Reply:
They are necessary in this case due to use of variables within case statements.